modules/pymol/commanding.py

```diff
@@ -657,36 +737,31 @@ def declare_command(name, function=None, _self=cmd):
         # Inner function that will be callable every time the command is executed
         @wraps(function)
         def inner(*args, **kwargs):
-            caller = traceback.extract_stack(limit=2)[0].filename
-
+            frame = traceback.format_stack()[-2]
+            caller = frame.split("\"", maxsplit=2)[1]
```

Comment:
Having all of this being called every time a command runs makes me a little nervous, especially if these commands are done in a loop. A lot of this info is just redundantly calculated?